### ISSUE TITLE
New Workspace Type

### ISSUE THREAD

**Issue by mhebrard** [awslabs/service-workbench-on-aws#386]
Hi, 

May I know the procedure to add a new workspace type ?

use-case:
I wish to add a custom cloudformation of EMR.
I noticed that cfn are defined in `addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/` but I am probably missing how to register the new template within the catalog and the UI

**Comment by SanketD92** [comment]
Hello @mhebrard, thank you for this question and your continued interest in exploring Service Workbench.

Please follow these steps for adding your custom CFN template:
1. Add your custom CloudFormation template in the `addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/service-catalog` folder
2. Add your new product's details in the format specified in the file `addons/addon-base-raas/packages/base-raas-post-deployment/lib/steps/create-service-catalog-portfolio.js` (lines 23-35).
3. Run the `environment-deploy.sh` script

I'll add a ticket in our backlog to add the documentation in a README file for this process.

**Comment by mhebrard** [comment]
Hi @SanketD92 Thanks for your help.

I updated the two files mentioned above, 
the deployment was successful.
but when I connect to workbench and navigate to "workspace types" I get that error:

![image](https://user-images.githubusercontent.com/1891334/111248554-1ea61b80-8645-11eb-8ead-e20127c56813.png)

I am wondering, should I modify something in `addons/addon-base-raas/packages/base-raas-cfn-templates/src/plugins/cfn-templates-plugin.js` ?

**Comment by SanketD92** [comment]
Hi @mhebrard, an update in the `addons/addon-base-raas/packages/base-raas-cfn-templates/src/plugins/cfn-templates-plugin.js` file would not be required.

Could you provide the following details for better debugging?

1. Does this error seem to be persistent, or does a refresh fix the issue?3. Could you confirm that the Service Catalog portfolio in your main account has the product you've added
2. Could you paste the error you see on the browser's network and console tabs when you load this page?

**Comment by SanketD92** [comment]
Hi @mhebrard, an update in the `addons/addon-base-raas/packages/base-raas-cfn-templates/src/plugins/cfn-templates-plugin.js` file would not be required.

Could you provide the following details for better debugging:

1. Does this error seem to be persistent, or does a refresh fix the issue?
2. Could you paste the error you see on the browser's network and console tabs when you load this page?
3. Could you confirm that the Service Catalog portfolio (main account's AWS management console) has the newly added product?

**Comment by mhebrard** [comment]
Hi @SanketD92,

Here some precisions:
1. the error persists even after refresh
2. Looking into it, it is a CORS error

```
(X) Access to fetch at 'https://[...].execute-api.ap-southeast-1.amazonaws.com/dev/api/workspace-type-candidates?status=not-imported&version=*' from origin 'https://[...].cloudfront.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
(X) GET https://[...].execute-api.ap-southeast-1.amazonaws.com/dev/api/workspace-type-candidates?status=not-imported&version=* net::ERR_FAILED
```

3. I confirmed that the Service Catalog portfolio has a new product as entended.

Thanks

**Comment by jn1119** [comment]
Thanks for verifying that information. The CORS error might point to an authorization error. Could you verify if you are signed in as an admin role?

Thanks

**Comment by jn1119** [comment]
Could you also verify if the Service Workbench "workspace types" tab works if you disassociate your newly created product from the portfolio. I am trying to see if the new product is the one that is causing an issue here. It would be helpful if you can post all the changes you made locally in order for us to reproduce and debug the issue. Thanks.

**Comment by mhebrard** [comment]
Hmm 
* yes, I am signed in as admin
* yes, when I dissociate the new prodict from the portfolio, the issue is solved

Trying to narrow down the problem, I limitted the changes to:
* an exact copy of the workbench EMR v1 cfn, with only the text in description changed, saved in
`/addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/service-catalog/grids-emr-cluster.cfn.yml`
* the addition of the new template in product to create:
```
// in addons/addon-base-raas/packages/base-raas-post-deployment/lib/steps/create-service-catalog-portfolio.js
const productsToCreate = [
// Add EMR v2 - GRIDS recipe
  {
    displayName: 'EMR from GRIDS',  // Please make sure the display name is unique in this list
    filename: 'grids-emr-cluster', // This is the name of your file you've added in the step above
    description: `An Amazon EMR research workspace that comes with: \n* Hail 0.2 \n* Jupyter Lab \n* Spark 2.4.4 \n* Hadoop 2.8.5`,
  },
// [...]
```

With this minimal changes, I am able to see the workspace types as expected with a new card for my new EMR type, and proceed through the test launch.

I guess the above error was due to syntax problem in the CFN
